src/noit_check_lmdb.c

```diff
@@ -42,15 +42,21 @@ static int noit_check_lmdb_show_check_json(mtev_http_rest_closure_t *restc,
 
 static int noit_check_lmdb_add_attribute(xmlNodePtr root, xmlNodePtr attr, noit_lmdb_check_data_t *key_data, MDB_val mdb_data, bool separate_stanza) {
   mtevAssert(root != NULL);
-  if ((!mdb_data.mv_data) || (mdb_data.mv_size == 0)) {
-    return 0;
-  }
+  char *val = NULL;
   if (!strcmp(key_data->key, "uuid")) {
     /* This should be set separately */
     return 0;
   }
-  char *val = (char *)calloc(1, mdb_data.mv_size + 1);
-  memcpy(val, mdb_data.mv_data, mdb_data.mv_size);
+  if ((mdb_data.mv_data == NULL) && (mdb_data.mv_size > 0)) {
+    /* If data is null, but we have a size, something is wrong - just skip
+     * this key */
+    mtevL(mtev_error, "noit_check_lmdb_add_attribute: got null data for key with a size, skipping\n");
```

Comment:
If there is anything useful we can log here, like the name of the attribute, or anything else that can make this error actionable by an operator/user, that would be helpful.

Reply:
Fixed